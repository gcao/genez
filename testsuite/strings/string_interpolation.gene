# Test string interpolation

# Basic interpolation
(var name "World")
(println #"Hello, #{name}!")

# Number interpolation
(var count 42)
(println #"The answer is #{count}")

# Float interpolation  
(var pi 3.14159)
(println #"Pi is #{pi}")

# Multiple interpolations
(var first "Gene")
(var last "Language")
(println #"#{first} #{last}")

# Expression interpolation
(var x 10)
(var y 20)
(println #"#{x} + #{y} = #{(+ x y)}")

# Empty string parts
(println #"#{name}")

# No interpolation
(println #"Just a string")