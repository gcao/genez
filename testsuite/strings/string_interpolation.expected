Hello, World!
The answer is 42
Pi is 3.14159
Gene Language
10 + 20 = 30
World
Just a string